wizeline/wize-q-remix
# feat: POC for migration from mysql to postgresql

#### What does this PR do?
This pull request is a POC to migrate from mysql to postgresql, Also I don't think this should be merger until we are sure the app doesn't randomly crash or has major compatibility issues with postgresql.  

#### Where should the reviewer start?
  prisma/schema.prisma

#### How should this be manually tested?
***For this to work you will need a local instance of postgresql, run the initial migration, and insert at least one department and location into the database. 
1.- Run the app
2.- If prompted to login, log in into the application
3.- At the home page, click on the "Ask Question" button
4.- Select a department
5.- Select a location
6.- Write at least 14 characters at the text box
7.- Click on the "Ask" button
8.- Click on the "Submit" button
9.- Verify that the question is correctly displayed. 

#### Any background context you want to provide?
As for initial findings for this ticket:
Some of our database naming practices are not fully compliant with postgresql best practices, which could be bad for raw queries into the database. 
 We have to transition some data types that are supported by mysql but not by postgresql such as DateTime and Float values
All our migration history will be erased which could be a good thing 
We still have pending to figure out a way to migrate our data
We don't have yet a proper dev environment such as the docker container for mysql
More effort should be made to ensure the app works end to end, at the moment only the happy path was verified and fixed. 
Nevertheless to say, the integration test are going to fail since some of the fields aren't compatible with mysql. 


#### What are the relevant tickets?
Related to #240

#### Screenshots
(Screenshots of the feature if available.)
:camera:

https://github.com/wizeline/wize-q-remix/assets/97201781/aab7b027-d3ce-4476-9e43-ccafccff5d82



#### Questions
Please let me hear your concerns since i'm not very experienced at tickets like this one.
